File: learnLeetCode_remote/Day29/Part1/energy_redistribution.cpp
#include <bits/stdc++.h>
using namespace std;

long long min_moves_to_equalize(const vector<long long>& energy){
    int n = (int)energy.size();
    if(n==0) return 0;
    long long S = 0; for(auto v: energy) S += v;
    if(S % n != 0) return -1;
    long long avg = S / n;
    long long surplus = 0;
    for(auto v: energy) if(v > avg) surplus += (v - avg);
    return surplus;
}

int main(){
    vector<long long> a = {1,3,5};
    cout << min_moves_to_equalize(a) << "\n"; // 2 (unit transfers)
    return 0;
}

File: learnLeetCode/Day29/Part1/energy_redistribution.py
from typing import List

def min_moves_to_equalize(energy: List[int]) -> int:
    n = len(energy)
    if n == 0:
        return 0
    S = sum(energy)
    if S % n != 0:
        return -1
    avg = S // n
    surplus = 0
    for v in energy:
        if v > avg:
            surplus += v - avg
    return surplus

if __name__ == "__main__":
    print(min_moves_to_equalize([1,3,5]))  # 2 (unit transfers)
